include/bout/invert/laplacexy.hxx

```diff
@@ -49,7 +49,7 @@
  */
 class LaplaceXY {
  public:
-  LaplaceXY(Mesh *m, Options *opt = nullptr, const CELL_LOC = CELL_CENTRE) {
+  LaplaceXY(Mesh *m = mesh, Options *opt = nullptr, const CELL_LOC = CELL_CENTRE) {
```

Comment:
I think this should default to `nullptr` and then the initialisation statement can use the usual pattern of checking if it's nullptr and if so setting to the global mesh. Two motivations being 1) I think this is closer to what we do in most of the rest of the code (probably not consistently though), 2) we can in the future change that logic to do something else without changes to the header (i.e. without changing the interface).